tasks/service/async/src/main/java/org/projectnessie/nessie/tasks/async/ScheduledHandle.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.nessie.tasks.async;
+
+import java.util.concurrent.CompletionStage;
+
+public interface ScheduledHandle {
```

Comment:
This interface seems a bit overkill, since it's just a wrapped completion stage. I think we can get rid of it and adapt the 2 impls:

```java
// TasksAsync
CompletionStage<Void> schedule(Runnable runnable, Instant scheduleNotBefore);

// JavaPoolTasksAsync impl:
completable.whenComplete(
        (v, t) -> {
          if (t instanceof CancellationException) {
            future.cancel(true); // allow interruption of blocking tasks
          }
        });

// VertxTasksAsync impl:
completableFuture.whenComplete(
        (v, t) -> {
          if (t instanceof CancellationException) {
            vertx.cancelTimer(timerId); // won't interrupt blocking tasks!
          }
        });
```

Reply:
I'm not actually sure about using `CompletionStage` as a scheduled task's handle. `CompletionStage` itself does not have a `cancel()` method.

If we do `CompletionStage.toCompletableFuture().cancel()`, IMHO, the relationship of the `cancel()` call to the original `CompletableFuture` is very indirect and implementation specific to rely upon.

I'd prefer to stick with a custom `ScheduledHandle` and a dedicated `cancel()` method in it.